Add README to datapipes

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63982 Add README to datapipes**

Add a readme to `datapipes` for developer. This is can be a replacement of https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb

After this PR is landed, the README.md will be added to PyTorch Wiki

Differential Revision: [D30554198](https://our.internmc.facebook.com/intern/diff/D30554198)

ejguan committed:
Add README to datapipes

[ghstack-poisoned]

ejguan committed:
Update on "Add README to datapipes"


Add a readme to `datapipes` for developer. This is can be a replacement of https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb

After this PR is landed, the README.md will be added to PyTorch Wiki

[ghstack-poisoned]

ejguan committed:
Update on "Add README to datapipes"


Add a readme to `datapipes` for developer. This is can be a replacement of https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb

After this PR is landed, the README.md will be added to PyTorch Wiki

Differential Revision: [D30554198](https://our.internmc.facebook.com/intern/diff/D30554198)

[ghstack-poisoned]

VitalyFedyunin reviewed (commented):
Can we take down https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb by combining it into this md?

VitalyFedyunin reviewed (approved):
super nit: Maybe mention why __len__ is not implemented in CVS example

ejguan committed:
Update on "Add README to datapipes"


Add a readme to `datapipes` for developer. This is can be a replacement of https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb

After this PR is landed, the README.md will be added to PyTorch Wiki

Differential Revision: [D30554198](https://our.internmc.facebook.com/intern/diff/D30554198)

[ghstack-poisoned]

ejguan committed:
Update on "Add README to datapipes"


Add a readme to `datapipes` for developer. This is can be a replacement of https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes_tutorial_dev_loaders.ipynb

After this PR is landed, the README.md will be added to PyTorch Wiki

Differential Revision: [D30554198](https://our.internmc.facebook.com/intern/diff/D30554198)

[ghstack-poisoned]
